Permit filtering unassigned peers

EdouardVanbelle opened the pull request:
Please find a very small improvement in peers list: 

- permit a serial lookup on peers table
- display the peer's serial number (to avoid. new column I added it in the OS section:
<img width="1785" alt="Serial Number" src="https://github.com/user-attachments/assets/3859c5a7-42ab-4880-93a5-c2ceaa27922a">

- display of the serial number in the peer section


- capability to filter peers that belong to the "All" group only, this is useful to track non assigned peers 
![All group only](https://github.com/user-attachments/assets/1cae36c8-a0be-4b29-bcde-2a74431ae9ce)

heisbrot commented:
Hey @EdouardVanbelle, 

thanks for this PR! I like the idea of filtering by unassigned peers. 
However, we should look into having it in another position to keep the filters clean. 

I was thinking of either in the groups filter or having a dropdown with (All Peers, Online Peers, Offline Peers, Unassigned Peers) instead of the current buttons. What do you think?

EdouardVanbelle commented:
Hello @heisbrot interesting ! You are right the current case is a bit confusing

I am also thinking that `All`, `Online`, `Offline` filters complements the  group filtering (you can mix both)

To complete your suggestion the dropdown with a selector before the separator and a checkbox on unassigned peers:

```
 ( ) All Peers
 ( ) Online Peers
 ( ) Offline Peers
 -----
 [ ] unassigned peers
```

Another possibility is to keep `All`, `Online`, `Offline` buttons and complete the dropdown group like this:
```
  [ ] groupA 
  [ ] groupB
  [ ] groupC
  -----
  [ ] unassigned group
```

Right now I have this incoherent case: (I must implement an exclusivity)

<img width="442" alt="Capture d’écran 2024-12-21 à 00 57 41" src="https://github.com/user-attachments/assets/0a4c7334-223f-42a5-9e2a-a7000d3dd0b2" />

EdouardVanbelle commented:
Hello @heisbrot , I applied your suggestions, which brings: 

<img width="1088" alt="Capture d’écran 2024-12-22 à 23 58 31" src="https://github.com/user-attachments/assets/19c96675-f4c3-40bd-a928-b53169caf849" />
<img width="1085" alt="Capture d’écran 2024-12-22 à 23 58 46" src="https://github.com/user-attachments/assets/b76df781-4f15-4cc7-9fd3-e5d8670a7a50" />

Changes: 
* there is an exclusivity between `Unassigned peers` and other group selection to keep consistency on this selector
* _breaking change_: I have also removed the `All` group in the drop down as all peers are always in this group, this part is arguable, up to you :)
* the Group filter is hidden if the only one group `All` exists, arguable too :)
* I have added a function `overrideTableFilter()` in order to keep other columns selection and reduce code redundancy. I guess there is a smarter solution but I don't know this framework
* _breaking change_: take care that now the `Pending Approvals` button keeps other column filters (use of `overrideTableFilter()` function)
* _breaking change_: I have also swapped buttons' order to keep filters together, and the RowsPerPage later, still up to you :)
* the serial is still present in this merge request


🙏 all my appologies, this is the first time I code in React

EdouardVanbelle commented:
Hello @heisbrot I have rebased these commit to resolve a conflict 
let me know if you prefer I split this merge request in 2:
1 for serial
1 for the filtering capability on unassigned peers

Regards

heisbrot commented:
> Hello @heisbrot I have rebased these commit to resolve a conflict let me know if you prefer I split this merge request in 2: 1 for serial 1 for the filtering capability on unassigned peers
> 
> Regards

Hey @EdouardVanbelle ,

thank you! Yes a split would be great. We can then directly merge the serial. 
The filtering stuff still needs review and some testing from my end. But overall the changes make sense :)

EdouardVanbelle committed:
peers table: refactor filters

EdouardVanbelle commented:
Here it is @heisbrot 

you can review first the pull request https://github.com/netbirdio/dashboard/pull/444 to validate the serial feature 
(I did not squashed your patch)

Then this pull request will contain only the `unassigned peers` filter

EdouardVanbelle committed:
add filtering on unassigned peers

EdouardVanbelle commented:
Hello @heisbrot if it helps the review, I split the commit in 2:
- one for the table filter refactor
- the other one for the filtering on unassigned peers

EdouardVanbelle commented:
Hello @heisbrot, let me know if still interested in this request and I will resolve the conflict :)
at least the first patch can interest you to refactor the filtering
